components/_global-navigation.scss

```diff
@@ -193,12 +210,24 @@ $notification-counter-size: 18px;
 		flex: 0 1 auto;
 	}
 
+	&__search {
+		flex: 0 1 auto;
+		transition: flex $search-transition-duration;
+	}
+
 	&__search-input-wrapper {
-		border: 0 solid $color-slate-gray;
+		border: {
+			bottom: 1px solid transparent;
+			left: 0;
+			right: 0;
+			top: 0;
+		};
```

Comment:
This way we set unnecessary values for all other borders then bottom one. And we immediately overwrite a value that's needed only for the bottom border. I'm not (yet 😉 ) convinced to your proposed solution.